HeterogeneousCore/SonicTriton/src/TritonData.cc

```diff
@@ -101,85 +119,151 @@ void TritonData<IO>::setBatchSize(unsigned bsize) {
     fullShape_[0] = batchSize_;
 }
 
+template <typename IO>
+void TritonData<IO>::computeSizes() {
+  sizeShape_ = sizeShape();
+  byteSizePerBatch_ = byteSize_ * sizeShape_;
+  totalByteSize_ = byteSizePerBatch_ * batchSize_;
+}
+template <typename IO>
+void TritonData<IO>::resetSizes() {
+  sizeShape_ = 0;
+  byteSizePerBatch_ = 0;
+  totalByteSize_ = 0;
+}
+
+//create a memory resource if none exists;
+//otherwise, reuse the memory resource, resizing it if necessary
+template <typename IO>
+bool TritonData<IO>::updateMem(size_t size, bool canThrow) {
+  bool status = true;
+  if (!memResource_ or size > memResource_->size()) {
```

Comment:
The size is determined by the batch size and the shape parameters; the calculation is, by design, exactly the same whether heap memory or shared memory is used. In principle, a "crazy" amount of memory could be requested, but it would be the result of a bug or user error.